### ISSUE TITLE
Aliasing implicit join paths produces invalid queries

### ISSUE THREAD

**Issue by SebastianS90** [jOOQ/jOOQ#12210]
### Expected behavior
The code
```java
Book b = BOOK.as("b");
Author a = b.author().as("a");
ctx.select(a.FIRST_NAME, a.LAST_NAME, b.TITLE).from(b)
```
should generate a query like
```sql
select "b".*, "a"."FIRST_NAME", "a"."LAST_NAME" from ("BOOK" "b" join "AUTHOR" "a" on "b"."AUTHOR_ID" = "a"."ID")
```

### Actual behavior
It generates
```sql
select "b".*, "a"."FIRST_NAME", "a"."LAST_NAME" from "BOOK" "b"
```
which is invalid, because the `AUTHOR` table is missing and `a` being undefined.

Without the `a` alias, e.g.
```diff
- Author a = b.author().as("a");
+ Author a = b.author();
```
the resulting query is
```sql
select "b".*, "alias_8277573"."FIRST_NAME", "alias_8277573"."LAST_NAME" from ("BOOK" "b" join "AUTHOR" "alias_8277573" on "b"."AUTHOR_ID" = "alias_8277573"."ID")
```
This is valid, but reading and debugging logged queries gets very frustrating if multiple such random alias names are produced.

Maybe related to #12068 (there you state that users can't provide aliases for implicit join paths).

### Versions

- jOOQ: 3.15.1
- Java: 11

**Comment by lukaseder** [comment]
Thanks a lot for your report. I thought this was fixed already, but it is not. See https://github.com/jOOQ/jOOQ/issues/12037 "Explicit path aliasing"

**Comment by lukaseder** [comment]
Luckily, this seems easier to fix than I would have thought.

**Comment by lukaseder** [comment]
Fixed for jOOQ 3.16.0, 3.15.3 (https://github.com/jOOQ/jOOQ/issues/12403), and 3.14.15 (https://github.com/jOOQ/jOOQ/issues/12404)

**Comment by lukaseder** [comment]
A related issue is:

- https://github.com/jOOQ/jOOQ/issues/12456
